routing/router.go

```diff
@@ -357,1510 +287,166 @@ type EdgeLocator struct {
 
-		}
+	quit chan struct{}
+	wg   sync.WaitGroup
+}
 
-		// If the channel is marked as a zombie in our database, and
-		// we consider this a stale update, then we should not apply the
-		// policy.
-		isStaleUpdate := time.Since(msg.LastUpdate) > r.cfg.ChannelPruneExpiry
-		if isZombie && isStaleUpdate {
-			return newErrf(ErrIgnored, "ignoring stale update "+
-				"(flags=%v|%v) for zombie chan_id=%v",
-				msg.MessageFlags, msg.ChannelFlags,
-				msg.ChannelID)
-		}
+// New creates a new instance of the ChannelRouter with the specified
+// configuration parameters. As part of initialization, if the router detects
+// that the channel graph isn't fully in sync with the latest UTXO (since the
+// channel graph is a subset of the UTXO set) set, then the router will proceed
+// to fully sync to the latest state of the UTXO set.
+func New(cfg Config) (*ChannelRouter, error) {
+	graph, err := NewCachedGraph(
+		cfg.SelfNode, cfg.RoutingGraph, false,
+	)
+	if err != nil {
+		return nil, err
+	}
 
-		// If the channel doesn't exist in our database, we cannot
-		// apply the updated policy.
-		if !exists {
-			return newErrf(ErrIgnored, "ignoring update "+
-				"(flags=%v|%v) for unknown chan_id=%v",
-				msg.MessageFlags, msg.ChannelFlags,
-				msg.ChannelID)
-		}
+	return &ChannelRouter{
+		cfg:         &cfg,
+		cachedGraph: graph,
+		quit:        make(chan struct{}),
+	}, nil
+}
 
-		// As edges are directional edge node has a unique policy for
-		// the direction of the edge they control. Therefore, we first
-		// check if we already have the most up-to-date information for
-		// that edge. If this message has a timestamp not strictly
-		// newer than what we already know of we can exit early.
-		switch {
-
-		// A flag set of 0 indicates this is an announcement for the
-		// "first" node in the channel.
-		case msg.ChannelFlags&lnwire.ChanUpdateDirection == 0:
-
-			// Ignore outdated message.
-			if !edge1Timestamp.Before(msg.LastUpdate) {
-				return newErrf(ErrOutdated, "Ignoring "+
-					"outdated update (flags=%v|%v) for "+
-					"known chan_id=%v", msg.MessageFlags,
-					msg.ChannelFlags, msg.ChannelID)
-			}
+// Start launches all the goroutines the ChannelRouter requires to carry out
+// its duties. If the router has already been started, then this method is a
+// noop.
+func (r *ChannelRouter) Start() error {
+	if !atomic.CompareAndSwapUint32(&r.started, 0, 1) {
+		return nil
+	}
 
-		// Similarly, a flag set of 1 indicates this is an announcement
-		// for the "second" node in the channel.
-		case msg.ChannelFlags&lnwire.ChanUpdateDirection == 1:
+	log.Info("Channel Router starting")
 
-			// Ignore outdated message.
-			if !edge2Timestamp.Before(msg.LastUpdate) {
-				return newErrf(ErrOutdated, "Ignoring "+
-					"outdated update (flags=%v|%v) for "+
-					"known chan_id=%v", msg.MessageFlags,
-					msg.ChannelFlags, msg.ChannelID)
-			}
-		}
+	// If any payments are still in flight, we resume, to make sure their
```

Comment:
Based on my other comment, I think it should start the new graph sub-system and block until it has caught up with the chain.